server/src/main/java/org/apache/druid/segment/realtime/FireDepartmentMetrics.java

```diff
@@ -261,7 +275,11 @@ public FireDepartmentMetrics snapshot()
     retVal.handOffCount.set(handOffCount.get());
     retVal.sinkCount.set(sinkCount.get());
     retVal.messageMaxTimestamp.set(messageMaxTimestamp.get());
+    retVal.maxSegmentHandoffTime.set(maxSegmentHandoffTime.get());
+    // Resetting to invalid value so that it's reported only once after it has been set.
+    maxSegmentHandoffTime.set(INVALID_SEGMENT_HANDOFF_TIME);
```

Comment:
Please don't interleave resetting of the current values with setting the values of the snapshot.
You can first prepare the snapshot, then reset the values separately (maybe in a separate method).

I wonder if some of the other metrics shouldn't be reset either. The count ones should be fine as we check the difference before reporting them.

Reply:
Maybe also add a test that demonstrates why the resetting is important.

Reply:
done